File: web/src/api/admin/system_config.js
import request from "../../request/index.js";

const baseUrl = '/v1/admin/system/config'

export const apiAdminSystemConfigBase = (data) => {
    return request.request({
        url: `${baseUrl}/base`,
        method: 'POST',
        data
    })
}

export const apiAdminSystemConfigGost = (data) => {
    return request.request({
        url: `${baseUrl}/gost`,
        method: 'POST',
        data
    })
}

export const apiAdminSystemConfigEmail = (data) => {
    return request.request({
        url: `${baseUrl}/email`,
        method: 'POST',
        data
    })
}

export const apiAdminSystemConfigEmailVerify = (data) => {
    return request.request({
        url: `${baseUrl}/emailVerify`,
        method: 'POST',
        data: data
    })
}

export const apiAdminSystemConfigQuery = (data) => {
    return request.request({
        url: `${baseUrl}/query`,
        method: 'POST',
        data
    })
}

export const apiAdminSystemConfigHome = (data) => {
    return request.request({
        url: `${baseUrl}/home`,
        method: 'POST',
        data
    })
}

File: server/router/admin/system_config.go
package admin

import (
	"github.com/gin-gonic/gin"
	"server/controller/admin/system_config"
	"server/global"
	"server/router/middleware"
)

func InitSystemConfig(group *gin.RouterGroup) {
    g := group.Group("system/config", middleware.Auth(global.Jwt), middleware.AuthAdmin())
    g.POST("base", system_config.Base)
    g.POST("gost", system_config.Gost)
    g.POST("email", system_config.Email)
    g.POST("emailVerify", system_config.EmailVerify)
    g.POST("query", system_config.Query)
    g.POST("home", system_config.Home)
}
